reference/test/distributed/partition_kernels.cpp

```diff
@@ -33,6 +33,22 @@ void assert_equal_data(const T* data, std::initializer_list<U> reference_data)
 }
 
 
+template <typename T, typename U>
+void assert_equal_segmented_array(const gko::segmented_array<T>& data,
```

Comment:
Maybe worth moving this to some header file for reuse across tests ? Or maybe even into a macro similar to `GKO_ASSERT_EQUAL_ARRAYS` ?

Reply:
for now this is only necessary in exactly one file, so I wouldn't put it in a header.